File: bin/sp-4.4.1/sicstus-4.4.1/se/sics/prologbeans/PBVariable.java
/**
 * Copyright (c) 2003 SICS AB. All rights reserved.
 */
package se.sics.prologbeans;

import java.io.IOException;

/**
 * <code>PBVariable</code> is the
 * representation of Prolog variables.
 */

class PBVariable extends PBTerm {
	/**
	 * Creates a new <code>PBVariable</code> instance with the specified name.
	 */
	PBVariable(String name) {
		super(name);
	}

	int getType() {
		return PBAtomic.VARIABLE;
	}

	public boolean isVariable() {
		return true;
	}

	String toPrologString() {
		return toString();
	}

	public String toString() {
		return name;
	}

	void fastWrite(FastWriter writer) throws IOException {
		writer.writeVariable(this);
	}

	// @Override
	PBTerm[] fastWritePrefix(FastWriter writer) throws IOException {
		writer.writeVariable(this);
		return PBTerm.NO_TERMS;
	}
}

File: bin/sp-4.4.1/sicstus-4.4.1/library/prologbeans.NET/PBVariable.cs

namespace se.sics.prologbeans
{
	
	/// <summary> <code>PBVariable</code> is the
	/// representation of Prolog variables.
	/// </summary>
	
	class PBVariable:PBTerm
	{
		virtual internal int Type {
			get {
				return PBAtomic.VARIABLE;
			}			
		}

		override public sealed bool Variable {
			get {
				return true;
			}			
		}

		/// <summary> Creates a new <code>PBVariable</code> instance with the specified name.</summary>
		internal PBVariable (string name) : base (name)
		{
		}

		internal override string toPrologString ()
		{
			return ToString ();
		}

		public override string ToString ()
		{
			return Name;
		}

		internal override PBTerm[] fastWritePrefix (FastWriter writer)
		{
			writer.writeVariable (this);
			return PBTerm.NO_TERMS;
		}
	}
}